src/commonMain/kotlin/fr/acinq/lightning/channel/states/Normal.kt

```diff
@@ -345,9 +359,99 @@ data class Normal(
+                                            }
+                                            Pair(this@Normal.copy(spliceStatus = SpliceStatus.None), actions)
+                                        } else if (spliceStatus.command.spliceOut?.scriptPubKey?.let { Helpers.Closing.isValidFinalScriptPubkey(it, allowAnySegwit = true) } == false) {
+                                            logger.warning { "cannot do splice: invalid splice-out script" }
+                                            spliceStatus.command.replyTo.complete(ChannelCommand.Commitment.Splice.Response.Failure.InvalidSpliceOutPubKeyScript)
+                                            val actions = buildList {
+                                                add(ChannelAction.Message.Send(Warning(channelId, InvalidSpliceRequest(channelId).message)))
+                                                add(ChannelAction.Disconnect)
+                                                // pending settlement commands will be replayed on reconnect
+                                            }
+                                            Pair(this@Normal.copy(spliceStatus = SpliceStatus.None), actions)
+                                        } else {
+                                            val spliceInit = SpliceInit(
+                                                channelId,
+                                                fundingContribution = fundingContribution,
+                                                lockTime = currentBlockHeight.toLong(),
+                                                feerate = spliceStatus.command.feerate,
+                                                fundingPubkey = channelKeys().fundingPubKey(parentCommitment.fundingTxIndex + 1),
+                                                pushAmount = spliceStatus.command.pushAmount
+                                            )
+                                            logger.info { "initiating splice with local.amount=${spliceInit.fundingContribution} local.push=${spliceInit.pushAmount}" }
+                                            Pair(this@Normal.copy(spliceStatus = SpliceStatus.Requested(spliceStatus.command, spliceInit)), listOf(ChannelAction.Message.Send(spliceInit)))
+                                        }
+                                    } else {
+                                        logger.warning { "cannot initiate splice, channel not quiescent" }
+                                        spliceStatus.command.replyTo.complete(ChannelCommand.Commitment.Splice.Response.Failure.ChannelNotQuiescent)
+                                        val actions = buildList {
+                                            add(ChannelAction.Message.Send(Warning(channelId, InvalidSpliceNotQuiescent(channelId).message)))
+                                            add(ChannelAction.Disconnect)
+                                            // pending settlement commands will be replayed on reconnect
```

Comment:
nit: remove (and everywhere else)

Reply:
fixed in [982d5d2](https://github.com/ACINQ/lightning-kmp/pull/558/commits/982d5d22fa484ef7be1bb7925664d50f32a4ca4b).